src/serialization.jl

```diff
@@ -209,11 +209,22 @@ function deserialize(::Type{T}, data::Dict) where {T <: NamedTuple}
     return T(key = data[string(key)] for key in fieldnames(T))
 end
 
-function deserialize(
-    ::Type{T},
-    data::Union{Nothing, Dict},
-) where {T <: Union{Nothing, NamedTuple}}
-    return isnothing(data) ? nothing : deserialize(T.b, data)
+# Some types that definitely won't be deserialized from Dicts
+const _NOT_FROM_DICT = Union{Nothing, Real, String}
```

Comment:
I'm not sure if it's ever possible to get another type of string, but we might want to use`AbstractString` instead.